src/main/java/uk/gov/companieshouse/api/accounts/service/impl/DebtorsService.java

```diff
@@ -134,6 +134,24 @@ public DebtorsService(DebtorsRepository repository,
         return new ResponseObject<>(ResponseStatus.FOUND, transformer.transform(entity));
     }
 
+    @Override
+    public ResponseObject<Debtors> deleteById(String id, HttpServletRequest request) throws DataException {
+        if (repository.existsById(id)) {
```

Comment:
I would move this statement to the `try and catch` as `repository.existsById(id)` throws a MongoException

```
try {
    if (repository.existsById(id)) {
        repository.deleteById(id);
        return new ResponseObject<>(ResponseStatus.UPDATED);
    } else {
        return new ResponseObject<>(ResponseStatus.NOT_FOUND);
    }

} catch (MongoException me) {
    ...
}
```